### ISSUE TITLE
`datamodel-codegen` crashes with SARIF 2.1.0 jsonschema

### ISSUE THREAD

**Issue by envp** [koxudaxi/datamodel-code-generator#1692]
**Describe the bug**
Running datamodel-code-generator in a virtual environment with offical sarif 2.1.0 schema crashes while trying to call `get` on a `bool` object.

**To Reproduce**

1. Set up a virtual environment (`python >= 3.9`), and activate it
2. Retrieve sarif 2.1.0 schema taken from official link into current dir

```sh
$ curl https://raw.githubusercontent.com/oasis-tcs/sarif-spec/main/Schemata/sarif-schema-2.1.0.json -O
```
3. Install datamodel-code-generator using pip in virtual env: `pip install -U datamodel-code-generator`

4. Run `datamodel-codegen` as: `datamodel-codegen --input sarif-schema-2.1.0.json --output sarif_model.py`

Used commandline:
```sh
$ datamodel-codegen --input sarif-schema-2.1.0.json --output sarif_model.py```
```

**Expected behavior**
Expected a file called `sarif_model.py` to be created corresponding to types defined in `sarif-schema-2.1.0.json`

**Actual behaviour**
Command in step 4 produces the following error:

```
$ datamodel-codegen --input sarif-schema-2.1.0.json --output sarif_model.py
The input file type was determined to be: jsonschema
This can be specificied explicitly with the `--input-file-type` option.
Traceback (most recent call last):
  File "/home/vyenamandra/code/sarif_model/.venv/lib/python3.11/site-packages/datamodel_code_generator/__main__.py", line 388, in main
    generate(
  File "/home/vyenamandra/code/sarif_model/.venv/lib/python3.11/site-packages/datamodel_code_generator/__init__.py", line 435, in generate
    results = parser.parse()
              ^^^^^^^^^^^^^^
  File "/home/vyenamandra/code/sarif_model/.venv/lib/python3.11/site-packages/datamodel_code_generator/parser/base.py", line 1135, in parse
    self.parse_raw()
  File "/home/vyenamandra/code/sarif_model/.venv/lib/python3.11/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1603, in parse_raw
    self._parse_file(self.raw_obj, obj_name, path_parts)
  File "/home/vyenamandra/code/sarif_model/.venv/lib/python3.11/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1665, in _parse_file
    root_obj = JsonSchemaObject.parse_obj(raw)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vyenamandra/code/sarif_model/.venv/lib/python3.11/site-packages/typing_extensions.py", line 2360, in wrapper
    return arg(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/vyenamandra/code/sarif_model/.venv/lib/python3.11/site-packages/pydantic/main.py", line 1027, in parse_obj
    return cls.model_validate(obj)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vyenamandra/code/sarif_model/.venv/lib/python3.11/site-packages/pydantic/main.py", line 503, in model_validate
    return cls.__pydantic_validator__.validate_python(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vyenamandra/code/sarif_model/.venv/lib/python3.11/site-packages/datamodel_code_generator/parser/jsonschema.py", line 270, in __init__
    super().__init__(**data)
  File "/home/vyenamandra/code/sarif_model/.venv/lib/python3.11/site-packages/pydantic/main.py", line 164, in __init__
    __pydantic_self__.__pydantic_validator__.validate_python(data, self_instance=__pydantic_self__)
  File "/home/vyenamandra/code/sarif_model/.venv/lib/python3.11/site-packages/datamodel_code_generator/parser/jsonschema.py", line 194, in validate_exclusive_maximum_and_exclusive_minimum
    exclusive_maximum: Union[float, bool, None] = values.get('exclusiveMaximum')
                                                  ^^^^^^^^^^
AttributeError: 'bool' object has no attribute 'get'
```

**Version:**
 - OS: 64bit Red Hat Enterprise Linux (also replicated on Intel macOS)
 - Python version: `3.9`, `3.10`, `3.11`
 - datamodel-code-generator version: 23 (`0.23.0`)

**Additional context**
I haven't tried `3.12` because it isn't available to our distro yet.

**Comment by koxudaxi** [comment]
Thank you for creating the issue.
We have the pr to fix it 
https://github.com/koxudaxi/datamodel-code-generator/pull/1691

**Comment by koxudaxi** [comment]
@envp 
I have released a fixed version  [0.24.0](https://github.com/koxudaxi/datamodel-code-generator/releases/tag/0.24.0)
Thank you so much!!
